src/averageAges.js

```diff
@@ -55,7 +91,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const women = getPeopleBySex(people, 'f');
+  const men = getPeopleBySex(people, 'm');
+
+  const womenWithChildren = (onlyWithSon)
+    ? women.filter(woman => hasChildren(woman, men))
+    : women.filter(woman => hasChildren(woman, people));
```

Comment:
you can use a special variable to avoid logic duplication
```suggestion
  const sample = onlyWithSon ? getPeopleBySex(people, 'm') : people;
  const womenWithChildren = women.filter(woman => hasChildren(woman, sample));
```